src/index.ts

```diff
@@ -6,7 +6,18 @@ const suspectConstructorRx = /"(?:c|\\u0063)(?:o|\\u006[Ff])(?:n|\\u006[Ee])(?:s
 const JsonSigRx = /^["[{]|^-?\d[\d.]{0,14}$/;
 
 function jsonParseTransform (key: string, value: any): any {
-  if (key === "__proto__" || key === "constructor") {
+  if (key === "__proto__") {
+    return;
+  }
+  if (key === "constructor") {
+    if (typeof value !== "object") {
+      return value;
+    }
+    const hasPrototype = Object.prototype.hasOwnProperty.call(value, "prototype");
```

Comment:
Shall we simplify it to `value && (`"prototype" in value)`?

Reply:
Like that?

Reply:
Check last commit.